pkg/daemon/update.go

```diff
@@ -173,27 +173,38 @@ func (dn *Daemon) reconcilable(oldConfig, newConfig *mcfgv1.MachineConfig) *stri
 
 	// Passwd section
 
-	// we don't currently configure groups or users in place. we can't fix it if
-	// something changed here.
+	// we don't currently configure Groups in place. we don't configure Users except
+	// for setting/updating SSHAuthorizedKeys for the only allowed user "core".
+	// otherwise we can't fix it if something changed here.
 	if !reflect.DeepEqual(oldIgn.Passwd, newIgn.Passwd) {
 		if !reflect.DeepEqual(oldIgn.Passwd.Groups, newIgn.Passwd.Groups) {
 			msg := "Ignition Passwd Groups section contains changes"
 			return &msg
 		}
-		// check if the prior config is empty and that this is the first time running.
-		// if so, the SSHKey from the cluster config and user "core" must be added to machine config,.
 		if !reflect.DeepEqual(oldIgn.Passwd.Users, newIgn.Passwd.Users) {
+			// check if the prior config is empty and that this is the first time running.
+			// if so, the SSHKey from the cluster config and user "core" must be added to machine config.
 			if len(oldIgn.Passwd.Users) == 0 && len(newIgn.Passwd.Users) == 1 {
-				if newIgn.Passwd.Users[0].Name == "core" && len(newIgn.Passwd.Users[0].SSHAuthorizedKeys) > 0 {
-					glog.Info("SSH Keys reconcilable")
-				} else {
-					msg := "Ignition passwd user section contains unsupported changes"
+				glog.Infof("user data sent to verify before ssh init: %v", newIgn.Passwd.Users[0])
+				msg := verifyUserFields(newIgn.Passwd.Users[0])
+				if msg != "" {
+					return &msg
+				}
+			} else if len(oldIgn.Passwd.Users) > 0 && len(newIgn.Passwd.Users) >= 1 {
+				// there is an update to Users, we must verify that it is ONLY making an acceptable
+				// change to the SSHAuthorizedKeys for the user "core"
+				for _, user := range newIgn.Passwd.Users {
+					if user.Name != "core" {
+						msg := "Ignition passwd user section contains unsupported changes: non-core user"
+						return &msg
+					}
+				}
+				glog.Infof("user data to be verified before ssh update: %v", newIgn.Passwd.Users[len(newIgn.Passwd.Users)-1])
+				msg := verifyUserFields(newIgn.Passwd.Users[len(newIgn.Passwd.Users)-1])
```

Comment:
Should this be `verifyUserFields(user)`?

Reply:
no, we are taking the last user in users in line with ignition logic.

Reply:
Wait sorry, I missed this was outside the `for` loop.